test/cpp/interop/interop_client.cc

```diff
@@ -948,6 +948,29 @@ bool InteropClient::DoCacheableUnary() {
   return true;
 }
 
+bool InteropClient::DoPickFirstUnary() {
+  const int rpcCount = 100;
+  SimpleRequest request;
+  SimpleResponse response;
+  std::string first_server_id;
+  request.set_fill_server_id(true);
+  for (int i = 0; i < rpcCount; i++) {
+    ClientContext context;
+    Status s = serviceStub_.Get()->UnaryCall(&context, request, &response);
+    if (!AssertStatusOk(s, context.debug_error_string())) {
+      return false;
+    }
+    if (i == 0) {
+      first_server_id = response.server_id();
+      gpr_log(GPR_DEBUG, "first_user_id is %s", first_server_id.c_str());
```

Comment:
nit: may be good to put this log after every call, so we can see the server id even in case of it not matching. E.g. something like:

```
gpr_log(GPR_DEBUG, "got server_id:%s", response.server_id().c_str());
if (i == 0) {
...
```

also s/first_user_id/first_server_id

Reply:
I think we can just call it `server_id`.
Instead of logging every response, maybe you can change the `GPR_ASSERT`  line 968 to something like
`if (response.server_id() != server_id) { gpr_log(GPR_ERROR, "%d rpc got server_id %s, expect id %s", i, response.server_id().c_str(), server_id.c_str());  return false;}`